src/KDSoapClient/KDSoapPendingCall.cpp

```diff
@@ -124,4 +111,15 @@ void KDSoapPendingCall::Private::parseReply()
         KDSoapMessageReader reader;
         reader.xmlToMessage(data, &replyMessage, 0, &replyHeaders);
     }
+
+    if (reply->error()) {
+        if(!replyMessage.isFault())
+        {
+            replyHeaders.clear();
+            replyMessage = KDSoapMessage();
+            replyMessage.setFault(true);
+            replyMessage.addArgument(QString::fromLatin1("faultcode"), QString::number(reply->error()));
```

Comment:
That's not very SOAP-1.2, is it? ;-)
Don't you need to update this to the 1.2 elements for faults (depending on the soap version set in KDSoapClientInterface), so this works with your other commit?

Reply:
yes, you are right, this is not very SOAP 1.2. I was just trying to make it work in my situation :-) 
What would be the best way to make the version available in this function? I ask this because it is very easy to create a circular dependency between KDSoapPendingCall and KDSoapClientInterface.

Reply:
One solution would be to pass it as int and cast it, but that's ugly.
A better solution is to set it directly into the private class, bypassing the public API:
http://www.kdab.com/~dfaure/2017/0001-Pass-SOAP-version-to-KDSoapPendingCall.patch
Feel free to integrate this patch into yours.
Thanks!

Reply:
I added a second commit to support the SOAP 1.2 faults. I added this to the message reader as well. And I added unit tests for both changes and both soap versions.

Reply:
Ah, you didn't push the updated commits to this merge request, but I found the commits in your branch. They look good, I'll merge them.

Hmm, if soap-1.2 says faults don't use http status code 500, then KDSoapServerSocket.cpp needs an update as well... Are you interested in fixing the server side too?